cap/worker_options.go

```diff
@@ -0,0 +1,120 @@
+package cap
+
+import (
+	"github.com/capatazlib/go-capataz/internal/c"
+)
+
+// Restart specifies when a goroutine gets restarted
+type Restart = c.Restart
+
+// Permanent specifies that a goroutine should be restarted whether or not there
+// are errors.
+//
+// You can specify this option using the WithRestart function
+var Permanent = c.Permanent
+
+// Transient specifies that a goroutine should be restarted if and only if the
+// goroutine failed with an error. If the goroutine finishes without errors it
+// is not restarted again.
+//
+// You can specify this option using the WithRestart function
+var Transient = c.Transient
+
+// Temporary specifies that the goroutine should not be restarted under any
+// circumstances
+//
+// You can specify this option using the WithRestart function
+var Temporary = c.Temporary
+
+// Shutdown is an enum type that indicates how the parent supervisor will handle
+// the stoppping of the worker goroutine
+//
+type Shutdown = c.Shutdown
+
+// Indefinitely is a Shutdown value that specifies the parent supervisor must
+// wait indefinitely for the worker goroutine to stop executing
+//
+// You can specify this option using the WithShutdown function
+var Indefinitely = c.Indefinitely
+
+// Timeout is a Shutdown function that returns a value that indicates the time
+// that the supervisor will wait before "force-killing" a worker goroutine. This
+// function receives a time.Duration value
+//
+// You can specify this option using the WithShutdown function
+//
+// * Warning
+//
+// Is important to emphasize that golang **does not** provide a "force-kill"
+// mechanism for goroutines.
+//
+// There is no known way to kill a goroutine via a signal other than using
+// context.Done which the supervised goroutine must observe and respect.
+//
+// If the timeout is reached and the goroutine does not stop (because the worker
+// goroutine is not using the offered context value), the supervisor will
+// continue with the shutdown procedure (reporting a shutdown error), possibly
+// leaving the goroutine running in memory (e.g. memory leak)
+var Timeout = c.Timeout
+
+// NodeTag specifies the type of node that is running, this is a closed set
```

Comment:
s/running, this/running. This/

It should not be this way, but the nomenclature change had me spinning for a while. Why is it it `NodeTag` and not `ChildTag`?